### ISSUE TITLE
TMS bad tileset configuration

### ISSUE THREAD

**Issue by RubMart** [mapproxy/mapproxy#304]
Configuring a TMS layer, the configuration of the Tilesets and therefore URLs based on a Grid Mercator is not correct, there is no real level 0, and causes any level is poorly coded. Actually, it lacks the base level of the resolution of Web Mercator, since the level that is indicated like 0, is actually the level of zoom 1.

For example, the level 0 of a Grid Mercator, in units per pixel has the value 156,412 according to official sources: http://wiki.openstreetmap.org/wiki/Zoom_levels

The result obtained

<TileSet href = "http: // localhost: 8000 / tms / 1.0.0 / MICAPA / osm_grid / 0" units-per-pixel = "78271.516964" order = "0" />

This causes the actual TMS addresses to be incorrect, using a lower than real zoom level (order). It should be level 15 to which this setting corresponds, but redirects to level 14.

<TileSet href = "http: // localhost: 8000 / tms / 1.0.0 / MICAPA / osm_grid / 14" units-per-pixel = "4.77731426782" order = "14" />

Http: // localhost: 8000 / tms / 1.0.0 / MICAPA / osm_grid / 14/16079 / 20408.png

And aditionally:
1) Not configuring min_resolution in TMS layer capabilities
2) Not configuring real layer BBOX in TMS layer capabilities

## Your Environment

* Version used: 1.9.0
* Environment name and version (e.g. Python 2.7.5 with mod_wsgi 4.5.9): Python 2.7.12
* Server type and version: debug-version
* Operating System and version: ubuntu 14.04 LTE
* Configuration of my YAML file:

sources:
  MICAPA:
    type: tile
    url: http://MYCUSTOMURI
    grid: osm_grid
    min_res: 9
    transparent: true
    coverage:
      bbox: [-3.380749,40.38536570,-3.345202,40.42151]
      bbox_srs: 'EPSG:4326'

grids:
    osm_grid:
        base: GLOBAL_MERCATOR
        srs: 'EPSG:3857'
        origin: sw

**Comment by olt** [comment]
The TMS specification defines that a webmercator tilegrid starts with four tiles (https://wiki.osgeo.org/wiki/Tile_Map_Service_Specification#global-mercator)

Please use ``/tiles`` instead of ``/tms`` to get the desired behavior. (See: https://mapproxy.org/docs/nightly/services.html#openlayers)

**Comment by RubMart** [comment]
And min and max resolution and real BBOX coverage for the layer in TMS Layer Capabilities that are set to default (world coverage and full resolution, tilesets from order 0 to 19 when are necesary only five or six) when both are configured in source layer. Example of TMS Layer Capabilites BBox:

`<BoundingBox minx="-20037508.3428" miny="-20037508.3428" maxx="20037508.3428"
maxy="20037508.3428" />`

**Comment by olt** [comment]
Yes, the capabilities could use the coverage for BoundingBox, as long as Offset is right.

**Comment by shimoncohen** [comment]
@weskamm I think this issue can be closed. Thank you very much!
